modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -144,7 +152,11 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, _ cadd
 	// to browse or return an error
 	if info.IsDir() {
 		if fsrv.Browse != nil && !fileHidden(filename, filesToHide) {
-			return fsrv.serveBrowse(filename, w, r)
+			if streamAsArchive {
+				return fsrv.streamFolderAsArchive(suffix, filename, contentType, w)
+			} else {
+				return fsrv.serveBrowse(filename, w, r)
+			}
```

Comment:
When this feature is moved into the Browse struct, this change will not be necessary; i.e. it will all happen within `serveBrowse()`.